[MC-15] - Create Password Input Component

pedroemanuel42 opened the pull request:
# Feature/Issue Description
This PR create a text input box for password with the functionality to hide or show the text written through the icon (eye icon) on the side.

# Changes Summary
- Add new scss archive for input box password;
- Add new fonts archives from google fonts;
- Add a new dependency from fontawesome;
- Add a new service for password input;
- Add a test for password input service.

# How Has This Been Tested?

- To test the code, I ran the application, went to components and validated the operation on localhost. 

# Images (Opcional)
![image](https://user-images.githubusercontent.com/99288851/159190374-99405ebc-85bb-4235-a8e3-d1f0f089b800.png)
![image](https://user-images.githubusercontent.com/99288851/159190388-f4e90acd-08a7-48eb-96e2-63c9bbd8fb61.png)

pedroemanuel42 committed:
MC15 update

Created a input button and added a font awesome link to use the lock icon on the placeholder, also created a password.scss where i'll customize this input.

pedroemanuel42 committed:
CSS updates

increment css attributes lefting only a few adjustments

pedroemanuel42 committed:
Updates 1.2

With the help of the architect, found somes issues and was guided to fix then, this commit was the intial part to solve

pedroemanuel42 committed:
Added Show/Hide Password icon

Added a function that's show and hide our password input. Lefting test and put the google fonts to the text.

pedroemanuel42 committed:
GoogleFonts added

pedroemanuel42 committed:
Reviwed code

Att variables, classes names and others stuffs to let the code more legible

pedroemanuel42 committed:
FIxed eye icon position

pedroemanuel42 committed:
Few updates in margin/padding

Adjusted a few things on margin and padding attributes, changed the input text color as well.

pedroemanuel42 committed:
Remove unecessary arq

pedroemanuel42 committed:
[Feature]MC-15 - overflow fixed

Fixed the overflow problem and a few adjusts to organize css (lefting only the import font-awesome problem)

pedroemanuel42 committed:
[Feature]MC-15

pedroemanuel42 committed:
[Feature]MC-15-css and test

An few adjusts to css files and added a first part of the test function initFor()

pedroemanuel42 committed:
[Feature]MC-15 Testing function

pedroemanuel42 committed:
[Feature]MC-15

pedroemanuel42 committed:
[Feature]MC-15 Updated CSS and tryied resolve the test.

-Changed HTML and CSS id's (underline for dash);

pedroemanuel42 committed:
[Feature]MC-15 Changes on PasswordInputService and CSS

Tasks done

-Added focus trigger to input;
-Changed input colour to $other-dark;
-Build the input test (lefting expect focus and icon change);

Tasks lefting

-Check if input and icon exists;
-Build test over this check up;
-Input on focus, change the colour icons to $other-light;

pedroemanuel42 committed:
[Feature]MC-15

-Add the icon change test;

pedroemanuel42 committed:
[Feature]MC-15

-Added an empty check for querys on initFor()

pedroemanuel42 committed:
[Feature]MC-15

-Changed log to throw error on initFor empty test

pedroemanuel42 committed:
[Feature]MC-15

-Added focus trigger test on passwordInputService

pedroemanuel42 committed:
[Feature]MC-15

-Changed errorMessage;
-Removed TypeError;

pedroemanuel42 committed:
[Feature]MC-15

-Add new hasQuery function to check if the function get the right input;

pedroemanuel42 committed:
[Feature]MC-15

-Fixed Added a Listener check to click event;
- Added an if condition for times that div, input and/or icon hasn't query.

pedroemanuel42 committed:
[Feature]MC-15

-Changed scss box password to style.scss on components

pedroemanuel42 committed:
[Feature] MC-15

- Create a function to check if queries has done;
- Add a test to this function;

pedroemanuel42 committed:
[Feature] MC-15

-Code formatation;
-Add an test for check input password type;

pedroemanuel42 committed:
[Feature] MC-15

- Remove function sum and his test;
- Remove one condition from checkQuery;
- Fix spyOn problem from build test;

RodrigoSoaresNascimento reviewed (dismissed):
is this sum function really necessary?

feelsantiago committed:
MC-15 Update path for codecov rule

tvps20 commented:
The confirmation message pattern is not being respected. As the delivery date is near, the demand will be considered, but not in the next.
[MC-task number message commit]

feelsantiago commented on src/assets/scss/input-password.scss:
```diff
@@ -0,0 +1,43 @@
+}
+
+// Icons
+#padlock-icon {
```
Should not use Id for this kind of component. Either query by assossiations tags or classes.

feelsantiago commented on src/assets/scss/input-password.scss:
```diff
@@ -0,0 +1,43 @@
+    left: 17px;
+}
+
+#eye-icon {
```
Same here.

feelsantiago commented on src/pages/components/main.js:
```diff
@@ -1,14 +1,12 @@
 import { LayoutProviderService } from '../../services/layout-provider.service';
+import { PasswordInputService } from '../../services/password-input.service';
 
+import '../../assets/scss/input-password.scss';
```
It's better to import this inside components/style.scss

feelsantiago commented on src/services/tests/password-input.service.test.js:
```diff
@@ -0,0 +1,126 @@
+        });
+
+        it('Should check if div has query', () => {
+            const div = `
```
Don't need this mock

feelsantiago commented on src/services/tests/password-input.service.test.js:
```diff
@@ -0,0 +1,126 @@
+            expect(resultInput.is(':focus')).toBe(true);
+        });
+
+        it('Should check if div has query', () => {
```
This description is confusing.

feelsantiago commented on src/services/tests/password-input.service.test.js:
```diff
@@ -0,0 +1,126 @@
+            }).toThrow('Failed to query input');
+        });
+
+        it('Should check if icon has query', () => {
```
Same here

pedroemanuel42 replied:
Done all things, thanks for the advices!

pedroemanuel42 committed:
MC-15 Request changes by feelssantiago

- Remove id icons and query then by classes;
- Remove import from components/main.js and added to components/style.scss;
- Remove unecessary mock on password-input.service.test.js;
- Change tests descriptions on password-input.service.test.js;

pedroemanuel42 commented:
Noted, thanks for the tips! @tvps20

pedroemanuel42 commented:
All done, thanks for the advices! @feelsantiago

pedroemanuel42 commented:
This was a test function, it was removed by @feelsantiago. @RodrigoSoaresNascimento

feelsantiago commented on src/services/tests/password-input.service.test.js:
```diff
@@ -0,0 +1,117 @@
+            expect(resultInput.is(':focus')).toBe(true);
+        });
+
+        it('Should throw error failed on query div', () => {
```
The description should make sanse. Like: Should throw error if query fail or not find element.

feelsantiago commented on src/services/tests/password-input.service.test.js:
```diff
@@ -0,0 +1,117 @@
+            }).toThrow('Failed to query div');
+        });
+
+        it('Should throw error failed on query input', () => {
```
Same here

feelsantiago commented on src/services/tests/password-input.service.test.js:
```diff
@@ -0,0 +1,117 @@
+            }).toThrow('Failed to query input');
+        });
+
+        it('Should throw error failed on query icon', () => {
```
Same

pedroemanuel42 committed:
MC-15 Request changes by feelssantiago

- Changed the tests descriptions when fail to query or don't find the elements.

pedroemanuel42 commented:
All done!

pedroemanuel42 committed:
Co-authored-by: Filipe Santiago <porto.santiago@hotmail.com>
